src/nvim/api/buffer.c

```diff
@@ -179,6 +179,11 @@ void buffer_set_line_slice(Buffer buffer,
     return;
   }
 
+  if (!inbounds(buf, start)) {
```

Comment:
I think that you might already be able to accomplish this using `set_line_slice(buf, -1, -1, false, true, ["lines", "to", "append"])`